src/diffusers/models/modeling_utils.py

```diff
@@ -883,12 +883,15 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         dduf_entries: Optional[Dict[str, DDUFEntry]] = kwargs.pop("dduf_entries", None)
         disable_mmap = kwargs.pop("disable_mmap", False)
 
-        if not isinstance(torch_dtype, torch.dtype):
+        if torch_dtype is not None and not isinstance(torch_dtype, torch.dtype):
             torch_dtype = torch.float32
             logger.warning(
                 f"Passed `torch_dtype` {torch_dtype} is not a `torch.dtype`. Defaulting to `torch.float32`."
             )
 
+        if quantization_config is not None and torch_dtype is not None:
```

Comment:
In this case, if we were trying to set the dtype of the model e.g. `FluxTransformer(quantization_config=BnBConfig, torch_dtype=torch.bfloat16)` Wouldn't the dtype be overwritten and then get set to the BnB default `float16`?

Reply:
Yes looks like it, I think it's ok without this part.